File: desktop/lib/dto/user_response.dart
class UserResponse {
  int id;
  String username;
  String email;
  bool isPremium;
  bool isAdmin;
  DateTime createdAt;

  UserResponse({
    required this.id,
    required this.username,
    required this.email,
    required this.isPremium,
    required this.isAdmin,
    required this.createdAt,
  });

  factory UserResponse.fromJson(Map<String, dynamic> json) => UserResponse(
        id: json["id"],
        username: json["username"],
        email: json["email"],
        isPremium: json["isPremium"],
        isAdmin: json["isAdmin"],
        createdAt: DateTime.parse(json["createdAt"]),
      );

  Map<String, dynamic> toJson() => {
        "id": id,
        "username": username,
        "email": email,
        "isPremium": isPremium,
        "isAdmin": isAdmin,
        "createdAt": createdAt.toIso8601String(),
      };
}

File: backend/MemoWise.Model/DTO/UserResponse.cs
namespace MemoWise.Model.DTO;

public class UserResponse
{
    public int Id { get; set; }
    public string Username { get; set; }
    public string Email { get; set; }
    public bool IsPremium { get; set; }
    public bool IsAdmin { get; set; }
    public DateTime CreatedAt { get; set; }
}
